rts/System/float3.h

```diff
@@ -422,6 +422,25 @@ class float3
 		};
 	}
 
+
+	/**
+	 * Rotate a vector by the angle between rotationVector and RgtVector.
+	 * The result is only normalized if the input vector and self are normalized.
+	 * @return new vector with the result
+	 */
+	float3 rotate2D(const float3& rotationVector) const {
+		// https://blog.demofox.org/2014/12/27/using-imaginary-numbers-to-rotate-2d-vectors/
+		float nx = x * rotationVector.x - z * rotationVector.z;
+		float nz = x * rotationVector.z + z * rotationVector.x;
+		return float3{ nx, y, nz };
+	}
+
```

Comment:
why is `rotate2D` here in the header but `snapToAxis` in the other file? also perhaps this one could be explicitly `inline`?

Reply:
`snapToAxis` uses `Sign` which is from `SpringMath` and the header didn't import it. Indeed, probably a good candidate for inlining.

Reply:
The other float3 functions aren't inlined, should I just let the compiler decide like the others do?

Reply:
yea if nothing is already inlined then it's probably fine not to introduce that